lib/src/models/multireddit.dart

```diff
@@ -3,13 +3,406 @@
+  /// If this information is not provided, will return null.
+  IconName get iconName => IconName.values.firstWhere(
+      (e) =>
+          e.toString() ==
+          ('IconName.' + _convertToCamelCase(_data['icon_name'])),
+      orElse: () => null);
+
+  List<SubredditRef> get subreddits {
+    final subredditList = [];
+    subredditList.addAll(_data['subreddits']
+        .map((subreddit) => new SubredditRef.name(reddit, subreddit['name'])));
+    return subredditList;
+  }
+
+  /// The [Map] of data associated with this multireddit.
+  Map get data => _data;
+  Map _data;
+
+  /// The [Redditor] associated with this multireddit.
+  RedditorRef get author => new RedditorRef.name(reddit, _author);
+  String _author;
+
+  /// The displayName given to the [Multireddit].
+  String get displayName => _data['display_name'];
+
+  /// The infoPath will be used to uniquely identify this multireddit.
+  String get infoPath => _infoPath ?? '/';
+  String _infoPath;
+
+  /// The visibility of this multireddit.
+  ///
+  /// Can be one of { hidden, private, public }.
+  /// If this information is not provided, will return null.
+  Visibility get visibility => Visibility.values.firstWhere(
+      (e) => e.toString() == ('Visibility.' + _data['visibility']),
+      orElse: () => null);
+
+  /// The visibility of this multireddit.
+  ///
+  /// Can be one of { hidden, private, public }.
+  /// If this information is not provided, will return null.
+  WeightingScheme get weightingScheme => WeightingScheme.values.firstWhere(
+      (e) => e.toString() == ('WeightingScheme.' + _data['weighting_scheme']),
+      orElse: () => null);
+
+  /// Does the currently authenticated [User] have the privilege to edit this
+  /// multireddit.
+  bool get canEdit => _data['can_edit'];
+
+  /// Does this multireddit require visitors to be over the age of 18.
+  bool get over18 => _data['over_18'];
+
   Multireddit.parse(Reddit reddit, Map data) : super(reddit) {
-    setData(this, data['data']);
+    _data = data['data'];
+    _author = _data['path']?.split('/')[_redditorNameInPathIndex];
+    _infoPath = _generateInfoPath(_data['name'], _author);
+  }
+
+  // Returns valid info_path for multireddit with name `name`.
+  static String _generateInfoPath(String name, String user) =>
+      apiPath['multireddit']
+          .replaceAll(_multiredditRegExp, name)
+          .replaceAll(_userRegExp, user);
+
+  // Converts input text into camelCase.
+  static String _convertToCamelCase(String str) {
+    if (str == null) {
+      return null;
+    }
+    final splitStrList = str.split(new RegExp(r'(\s|_)+'));
+    final strItr = splitStrList.iterator;
+    final strBuffer = new StringBuffer(strItr.toString());
+    while (strItr.moveNext()) {
+      strBuffer.write(strItr.toString().substring(0, 1).toUpperCase());
+      strBuffer.write(strItr.toString().substring(1));
+    }
+    return strBuffer.toString();
+  }
+
+  /// Returns a slug version of the [title].
```

Comment:
What exactly is a slug in this context?

Reply:
This is supposed to be a utility for creating the correct `display_name`.